test/eventinge2e/source_broker_ksvc_test.go

```diff
@@ -0,0 +1,118 @@
+	triggerKind       = "trigger"
+)
+
+func TestKnativeSourceBrokerTriggerKnativeService(t *testing.T) {
+	client := test.SetupClusterAdmin(t)
+	test.CleanupOnInterrupt(t, func() { test.CleanupAll(t, client) })
+	defer test.CleanupAll(t, client)
+
+	// Setup a knative service
+	ksvc, err := test.WithServiceReady(client, helloWorldService, testNamespace, image)
+	if err != nil {
+		t.Fatal("Knative Service not ready", err)
+	}
+
+	cm := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: cmName,
+		},
+		Data: map[string]string{
+			"channelTemplateSpec": fmt.Sprintf(`
+apiVersion: %q
+kind: %q`, channelAPIVersion, channelKind),
+		},
+	}
+	configMap, err := client.Clients.Kube.CoreV1().ConfigMaps(testNamespace).Create(cm)
+	if err != nil {
+		t.Fatal("Unable to create ConfigMap: ", err)
+	}
+	br := &eventingv1beta1.Broker{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      brokerName,
+			Namespace: testNamespace,
+		},
+		Spec: eventingv1beta1.BrokerSpec{
+			Config: &duckv1.KReference{
+				APIVersion: "v1",
+				Kind:       "ConfigMap",
+				Name:       configMap.Name,
+			},
+		},
+	}
+	broker, err := client.Clients.Eventing.EventingV1beta1().Brokers(testNamespace).Create(br)
+	if err != nil {
+		t.Fatal("Unable to create broker: ", err)
+	}
+	tr := &eventingv1beta1.Trigger{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      triggerName,
+			Namespace: testNamespace,
+		},
+		Spec: eventingv1beta1.TriggerSpec{
+			Broker: broker.Name,
+			Subscriber: duckv1.Destination{
+				Ref: &duckv1.KReference{
+					APIVersion: ksvcAPIVersion,
+					Kind:       ksvcKind,
+					Name:       helloWorldService,
+				},
+			},
+		},
+	}
+	trigger, err := client.Clients.Eventing.EventingV1beta1().Triggers(testNamespace).Create(tr)
+	if err != nil {
+		t.Fatal("Unable to create trigger: ", err)
+	}
+
+	ps := &eventingsourcesv1alpha2.PingSource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pingSourceName,
+			Namespace: testNamespace,
+		},
+		Spec: eventingsourcesv1alpha2.PingSourceSpec{
+			JsonData: helloWorldText,
+			SourceSpec: duckv1.SourceSpec{
+				Sink: duckv1.Destination{
+					Ref: &duckv1.KReference{
+						APIVersion: brokerAPIVersion,
+						Kind:       brokerKind,
+						Name:       broker.Name,
+					},
+				},
+			},
+		},
+	}
+	_, err = client.Clients.Eventing.SourcesV1alpha2().PingSources(testNamespace).Create(ps)
+	if err != nil {
+		t.Fatal("Knative PingSource not created: %+V", err)
+	}
+	waitForRouteServingText(t, client, ksvc.Status.URL.URL(), helloWorldText)
+
+	// Delete resources
+	client.Clients.Eventing.EventingV1beta1().Brokers(testNamespace).Delete(broker.Name, &metav1.DeleteOptions{})
+	client.Clients.Eventing.EventingV1beta1().Triggers(testNamespace).Delete(trigger.Name, &metav1.DeleteOptions{})
+	client.Clients.Eventing.SourcesV1alpha2().PingSources(testNamespace).Delete(ps.Name, &metav1.DeleteOptions{})
+	client.Clients.Kube.CoreV1().ConfigMaps(testNamespace).Delete(configMap.Name, &metav1.DeleteOptions{})
```

Comment:
Schedule a cleanup after every create call above to make sure stuff gets cleaned up even if the test is failed/aborted.

Or, as you know all the names beforehand anyway and we don't care about errors: Do this at the beginning of the test:

```golang
cleanup := func() {
	client.Clients.Eventing.EventingV1beta1().Brokers(testNamespace).Delete(broker.Name, &metav1.DeleteOptions{})
	client.Clients.Eventing.EventingV1beta1().Triggers(testNamespace).Delete(trigger.Name, &metav1.DeleteOptions{})
	client.Clients.Eventing.SourcesV1alpha2().PingSources(testNamespace).Delete(ps.Name, &metav1.DeleteOptions{})
	client.Clients.Kube.CoreV1().ConfigMaps(testNamespace).Delete(configMap.Name, &metav1.DeleteOptions{})
}
test.CleanupOnInterrupt(t, cleanup)
defer cleanup()
```

Reply:
done in `b4623e5` thanks for the suggestion